tests/pulsar/test_client.py

```diff
@@ -0,0 +1,70 @@
+# Tests adapted from Apache Pulsar tests:
+# https://github.com/apache/pulsar-client-python/blob/main/tests/pulsar_test.py
+import subprocess
+from unittest import main
+
+import pytest
+from _pulsar import Timeout
+
+from eodhp_utils.pulsar.client import Consumer, Producer
+
+timeout_ms = 2000  #
+service_url = "pulsar://localhost:6650"
+
+
+@pytest.fixture(autouse=True)
+def environment_variables(monkeypatch):
+    monkeypatch.setenv("PULSAR_URL", service_url)
+
+
+@pytest.fixture(scope="session")
+def run_pulsar_service():
+    subprocess.run(["build-support/pulsar-test-service-start.sh"])
+    yield
+    subprocess.run(["build-support/pulsar-test-service-stop.sh"])
+
+
+@pytest.mark.parametrize(
+    "message",
+    [b"Hello", "Hello", {"hello": "hello"}],
+)
+def test_producer(message, run_pulsar_service):
+    topic_name = "test_topic"
+    producer_name = "test_producer"
+
+    producer = Producer(topic=topic_name, producer=producer_name)
+    producer.send(message)
+
+    assert producer
```

Comment:
what is being tested here? Is line 46 tested at all? All that seems to be being tested is that the Producer constructor has not failed, e.g. that producer is not None.